### PR TITLE
Fix missing DLL in windows packaging

### LINKED ISSUE #1801: fix broken tray monitor on windows
the tray monitor is broken with the new cross-build container. Apparently `libpcre2-8-0.dll` is missing from packaging.

You can currently work around the issue by copying the DLL into your Bareos directory.
That file can be obtained from the Fedora 40 package `mingw64-pcre2-10.42-5.fc40.noarch.rpm`.

### PR DESCRIPTION
**Pull request by arogge**
After switching to a newer crossbuild-image a DLL for bareos-tray-monitor.exe is missing from packaging. This PR adds that DLL and fixes some minor issues with the NSI script.

This will resolve #1801 

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)
- [Boy Scout Rule](https://docs.bareos.org/DeveloperGuide/generaldevel.html#boy-scout-rule)

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
Make sure you check/merge the PR using `devtools/pr-tool` to have some simple automated checks run and a proper changelog record added.

##### General
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Required backport PRs have been created

##### Source code quality
- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR

### CODE REVIEW

File: core/platforms/win32/winbareos.nsi [hunk 1]
```diff
@@ -1496,15 +1497,17 @@ done:
   File "/oname=$PLUGINSDIR\libintl-8.dll" "libintl-8.dll"
   File "/oname=$PLUGINSDIR\libwinpthread-1.dll" "libwinpthread-1.dll"
 
-  # one of the two files  have to be available depending onwhat openssl Version we sue
-  File /nonfatal "/oname=$PLUGINSDIR\libcrypto-1_1.dll" "libcrypto-1_1.dll"
-  File /nonfatal "/oname=$PLUGINSDIR\libcrypto-1_1-x64.dll" "libcrypto-1_1-x64.dll"
-  # Either one of this two files will be available depending on 32/64 bits.
-  File /nonfatal "/oname=$PLUGINSDIR\libgcc_s_sjlj-1.dll" "libgcc_s_sjlj-1.dll"
-  File /nonfatal "/oname=$PLUGINSDIR\libgcc_s_seh-1.dll" "libgcc_s_seh-1.dll"
```
[the remark this answers is not in the document]
Comment (sebsura) [reply core/platforms/win32/winbareos.nsi:1504]: Seems like seh-1.dll is still part of the created installer.  Do you know why ?

**Review by sebsura — COMMENTED** [review]
I added some comments

File: core/platforms/win32/winbareos.nsi (hunk 1 above)
[the remark this answers is not in the document]
Comment (arogge) [reply core/platforms/win32/winbareos.nsi:1504]: This is "just" the copying of the dlls into $PLUGINSDIR.
I *think* I removed just the ones that weren't there when I built it (and would have been skipped with /nonfatal).
I'll recheck.

[the remark this answers is not in the document]
Comment (arogge) [reply core/platforms/win32/winbareos.nsi:1504]: I re-added the `libgcc_s_seh-1.dll` for 64-bits and added `libgcc_s_dw2-1.dll` for 32-bits, which was previously missing.

**Comment by arogge** [comment]
I addresses your remarks, I hope all is good now.

**Review by sebsura — APPROVED** [review]
Looks good to me.

**Pull request merged**